stdlib/testing/assert_empty.go

```diff
@@ -0,0 +1,126 @@
+package testing
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+)
+
+const AssertEmptyKind = "assertEmpty"
+
+type AssertEmptyOpSpec struct{}
+
+func (s *AssertEmptyOpSpec) Kind() flux.OperationKind {
+	return AssertEmptyKind
+}
+
+func init() {
+	assertEmptySignature := semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			flux.TablesParameter: flux.TableObjectType,
+		},
+		Required:     semantic.LabelSet{flux.TablesParameter},
+		Return:       flux.TableObjectType,
+		PipeArgument: flux.TablesParameter,
+	}
+
+	flux.RegisterPackageValue("testing", "assertEmpty", flux.FunctionValue(AssertEmptyKind, createAssertEmptyOpSpec, assertEmptySignature))
+	flux.RegisterOpSpec(AssertEmptyKind, newAssertEmptyOp)
+	plan.RegisterProcedureSpec(AssertEmptyKind, newAssertEmptyProcedure, AssertEmptyKind)
+	execute.RegisterTransformation(AssertEmptyKind, createAssertEmptyTransformation)
+}
+
+func createAssertEmptyOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+	return &AssertEmptyOpSpec{}, nil
+}
+
+func newAssertEmptyOp() flux.OperationSpec {
+	return new(AssertEmptyOpSpec)
+}
+
+type AssertEmptyProcedureSpec struct {
+	plan.DefaultCost
+}
+
+func (s *AssertEmptyProcedureSpec) Kind() plan.ProcedureKind {
+	return AssertEmptyKind
+}
+
+func (s *AssertEmptyProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := *s
+	return &ns
+}
+
+func newAssertEmptyProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	if _, ok := qs.(*AssertEmptyOpSpec); !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+	return &AssertEmptyProcedureSpec{}, nil
+}
+
+type AssertEmptyTransformation struct {
+	failures int64
+
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+}
+
+func createAssertEmptyTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	dataset := execute.NewDataset(id, mode, cache)
+	if _, ok := spec.(*AssertEmptyProcedureSpec); !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+
+	transform := NewAssertEmptyTransformation(dataset, cache)
+	return transform, dataset, nil
+}
+
+func NewAssertEmptyTransformation(d execute.Dataset, cache execute.TableBuilderCache) *AssertEmptyTransformation {
+	return &AssertEmptyTransformation{
+		d:     d,
+		cache: cache,
+	}
+}
+
+func (t *AssertEmptyTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *AssertEmptyTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	var failed bool
+	if err := tbl.Do(func(cr flux.ColReader) error {
```

Comment:
The table interface has an `Empty` method you can call directly

Reply:
We still need to call `Do` if I remember correctly. I wasn't sure if I should rely on `.Empty()`. I'll change things to use that.

Reply:
Hmm, I think you are right that we need to read the data or it will deadlock. Can you add a comment to that fact since it is easy to miss?